main.go

```diff
@@ -53,215 +60,346 @@ func NewClient(login, password string, httpClient *http.Client) *Client {
-		"org":        {"customer"},
-		"old-token":  {cli.login},
-		"IDToken2":   {cli.password},
-		"IDToken1":   {cli.uid},
+func (cli *Client) getDevices() DevicesInfo {
+	req, err := http.NewRequest(http.MethodGet, urlDevices, nil)
+	req.Header.Add("User-Agent", hUserAgent)
+	req.Header.Set("Accept", hContentTypeJson)
+	req.Header.Set("Authorization", basicAuthStr)
+	res, err := cli.httpClient.Do(req)
+	if err != nil {
+		log.Printf("req do err: %s", err)
 	}
+	defer res.Body.Close()
 
-	resp, err := cli.http.PostForm(
-		"https://login.yota.ru/UI/Login",
-		payload,
-	)
-	defer resp.Body.Close()
+	body, _ := ioutil.ReadAll(res.Body)
+	var jsonData map[string]interface{}
+	json.Unmarshal([]byte(body), &jsonData)
 
+	var devicesInfo DevicesInfo
+	err = json.Unmarshal(body, &devicesInfo)
 	if err != nil {
-		return err
-	}
-
-	redirected := resp.Request.URL.String()
-	if redirected != urlLoginSuccess {
-		return errors.New("redirected to not success url")
+		log.Println(err)
 	}
 
-	return nil
+	return devicesInfo
 }
 
-func (cli *Client) GetTariffs() ([]Tariff, error) {
-	resp, err := cli.http.Get(urlDevices)
-	defer resp.Body.Close()
-
+func (cli *Client) getTokenInfo() (err error) {
+	req, err := http.NewRequest(http.MethodPost, urlTokenInfo, nil)
+	req.Header.Add("User-Agent", hUserAgent)
+	req.Header.Set("Accept", hContentTypeJson)
+	res, err := cli.httpClient.Do(req)
 	if err != nil {
-		return nil, err
+		return
 	}
+	defer res.Body.Close()
+	return
+}
 
-	body, _ := ioutil.ReadAll(resp.Body)
-	matches := reTariffs.FindSubmatch(body)
-	if len(matches) == 0 {
-		return nil, errors.New("could not find raw data")
+func (cli *Client) getExecution() (execution string, err error) {
+	data := url.Values{
+		"client_id":     {"yota_mya"},
+		"client_secret": {"password"},
+		"realm":         {"/customer"},
+		"service":       {"dispatcher"},
+		"grant_type":    {"urn:roox:params:oauth:grant-type:m2m"},
+		"response_type": {"token cookie"},
 	}
-	rawData := matches[1]
 
-	decoded := map[string]map[string]interface{}{}
-	err = json.Unmarshal(rawData, &decoded)
+	req, err := http.NewRequest(http.MethodPost, urlAccessToken, strings.NewReader(data.Encode()))
+	req.Header.Add("User-Agent", hUserAgent)
+	req.Header.Set("Accept", hContentTypeJson)
+	req.Header.Set("Content-Type", hContentTypeForm)
+	res, err := cli.httpClient.Do(req)
 	if err != nil {
-		return nil, err
+		return "", err
 	}
+	defer res.Body.Close()
 
-	tariffs := []Tariff{}
-	for product, data := range decoded {
-		current := parseTariff(data["currentProduct"].(map[string]interface{}))
-		steps := data["steps"].([]interface{})
-
-		for _, step := range steps {
-			tariff := parseTariff(step.(map[string]interface{}))
-			if tariff.Code == current.Code {
-				tariff.Active = true
-			}
-
-			tariff.Product = product
+	body, _ := ioutil.ReadAll(res.Body)
+	var jsonData map[string]interface{}
+	json.Unmarshal([]byte(body), &jsonData)
+	execution = jsonData["execution"].(string)
```

Comment:
It doesn't look right that you use map[string]interface{} and then do a runtime assertion of a field execution. You should create a type like `ResponseExecution` and unmarshal the JSON into this struct. It's much safer than unmarshalling into a map[].